runtime-light/stdlib/crypto/crypto-functions.cpp

```diff
@@ -115,3 +110,175 @@ task_t<int64_t> f$openssl_verify(const string &data, const string &signature, co
 
   co_return 1;
 }
+
+Optional<tl::CipherAlgorithm> parse_algorithm(const string &method) {
+  if (strncasecmp(method.c_str(), "aes-128-cbc", strlen("aes-128-cbc")) == 0) {
+    return tl::CipherAlgorithm::AES128;
+  } else if (strncasecmp(method.c_str(), "aes-256-cbc", strlen("aes-256-cbc")) == 0) {
+    return tl::CipherAlgorithm::AES256;
+  }
+  return {};
+}
+
+int64_t algorithm_iv_len(tl::CipherAlgorithm algorithm __attribute__((unused))) {
+  /* since only aes-128/256-cbc supported for now */
+  return 16;
+}
+
+int64_t algorithm_key_len(tl::CipherAlgorithm algorithm) {
+  switch (algorithm) {
+    case tl::CipherAlgorithm::AES128: {
+      return 16;
+    }
+    case tl::CipherAlgorithm::AES256: {
+      return 32;
+    }
+  }
+}
+
+array<string> f$openssl_get_cipher_methods(bool aliases __attribute__((unused))) noexcept {
+  array<string> return_value{{std::make_pair(0, string("aes-128-cbc")), std::make_pair(1, string("aes-256-cbc"))}};
+  return return_value;
+}
+
+Optional<int64_t> f$openssl_cipher_iv_length(const string &method) noexcept {
+  auto algorithm = parse_algorithm(method);
+  if (algorithm.is_null()) {
+    php_warning("Unknown cipher algorithm");
+    return false;
+  }
+  return algorithm_iv_len(algorithm.val());
+}
+
+enum cipher_opts : int { OPENSSL_RAW_DATA = 1, OPENSSL_ZERO_PADDING = 2, OPENSSL_DONT_ZERO_PAD_KEY = 4 };
```

Comment:
1. Let's move non-public symbols into an anonymous namespace
2. Let's also either use `enum class` or distinct constants since I don't see any value from using `enum` here.
3. We also try to avoid implicit types like `int`

Reply:
fix